### PR TITLE
Lower gas limit and make it updateable

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Lower gas limit
- Make the gas limit updateable

### CODE REVIEW

**Comment by alexkeating** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #94** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting/94?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting/94?utm_source=stack-comment).

File: src/WormholeSender.sol [hunk 1]
```diff
@@ -24,4 +24,10 @@ abstract contract WormholeSender is WormholeBase {
   function quoteDeliveryCost(uint16 targetChain) public view virtual returns (uint256 cost) {
     (cost,) = WORMHOLE_RELAYER.quoteEVMDeliveryPrice(targetChain, 0, GAS_LIMIT);
   }
+
+  /// @param gasLimit The new gas limit value which is used to estimate the delivery cost for
+  /// sending messages cross chain.
+  function updateGasLimit(uint256 gasLimit) public onlyOwner {
+    GAS_LIMIT = gasLimit;
```
Comment (wildmolasses) [inline src/WormholeSender.sol:31]: let's add an event and make the param `_gasLimit`

File: src/WormholeSender.sol [hunk 2]
```diff
@@ -11,7 +11,7 @@ abstract contract WormholeSender is WormholeBase {
   uint16 public immutable REFUND_CHAIN;
 
   /// @notice The gas limit for cross chain transactions.
-  uint256 constant GAS_LIMIT = 500_000;
+  uint256 public GAS_LIMIT = 200_000;
```
Comment (wildmolasses) [inline src/WormholeSender.sol:14]: var is no longer constant

```suggestion
  uint256 public gasLimit = 200_000;
```

File: test/WormholeBase.t.sol [hunk 3]
```diff
@@ -8,7 +8,7 @@ import {TestConstants} from "test/Constants.sol";
 
 contract Constructor is Test, TestConstants {
   function testForkFuzz_CorrectlySetsAllArgs(address _wormholeRelayer) public {
-    WormholeBase base = new WormholeBase(_wormholeRelayer);
+    WormholeBase base = new WormholeBase(_wormholeRelayer, msg.sender);
```
Comment (wildmolasses) [inline test/WormholeBase.t.sol:11]: can we fuzz the owner and assert it's set correctly?

can we also make a new test that makes sure only the owner can change the owner, and not a _notOwner address?

File: test/WormholeSender.t.sol [hunk 4]
```diff
@@ -61,3 +61,10 @@ contract QuoteDeliveryCost is WormholeSenderTest {
     assertGt(cost, 0, "No cost was quoted");
   }
 }
+
+contract UpdateGasLimit is WormholeSenderTest {
+  function testFuzz_CorrectlyUpdateGasLimit(uint256 gasLimit) public {
+    wormholeSender.updateGasLimit(gasLimit);
+    assertEq(wormholeSender.GAS_LIMIT(), gasLimit, "The gas limit is incorrect");
```
Comment (wildmolasses) [inline test/WormholeSender.t.sol:68]: can we add an expectEmit, and also add a test to make sure _notOwner can't call this method?

**Review by wildmolasses — CHANGES_REQUESTED** [review]
a few small things

File: src/WormholeSender.sol (hunk 1 above)
Comment (alexkeating) [reply src/WormholeSender.sol]: https://github.com/ScopeLift/l2-flexible-voting/pull/94/commits/5fb57cdad6d3b0bce2cf8f63c6c873ef27f761ae

File: src/WormholeSender.sol (hunk 2 above)
Comment (alexkeating) [reply src/WormholeSender.sol]: https://github.com/ScopeLift/l2-flexible-voting/pull/94/commits/5fb57cdad6d3b0bce2cf8f63c6c873ef27f761ae

File: test/WormholeSender.t.sol (hunk 4 above)
Comment (alexkeating) [reply test/WormholeSender.t.sol]: https://github.com/ScopeLift/l2-flexible-voting/pull/94/commits/5fb57cdad6d3b0bce2cf8f63c6c873ef27f761ae

File: test/WormholeBase.t.sol (hunk 3 above)
Comment (alexkeating) [reply test/WormholeBase.t.sol]: https://github.com/ScopeLift/l2-flexible-voting/pull/94#discussion_r1486860753

Comment (alexkeating) [reply test/WormholeBase.t.sol]: https://github.com/ScopeLift/l2-flexible-voting/commit/5fb57cdad6d3b0bce2cf8f63c6c873ef27f761ae

File: test/WormholeSender.t.sol [hunk 5]
```diff
@@ -61,3 +63,19 @@ contract QuoteDeliveryCost is WormholeSenderTest {
     assertGt(cost, 0, "No cost was quoted");
   }
 }
+
+contract UpdateGasLimit is WormholeSenderTest {
+  function testFuzz_CorrectlyUpdateGasLimit(uint256 gasLimit) public {
+    vm.expectEmit();
+    emit GasLimitUpdate(200_000, gasLimit, address(this));
```
Comment (wildmolasses) [inline test/WormholeSender.t.sol:70]: nit

```suggestion
    emit GasLimitUpdate(wormholeSender.gasLimit(), gasLimit, address(this));
```

**Review by wildmolasses — APPROVED** [review]
lgtm, one nit, feel free to ignore

**Pull request merged**